rust/relay/ebpf-turn-router/src/main.rs

```diff
@@ -1,82 +1,42 @@
 #![cfg_attr(target_arch = "bpf", no_std)]
 #![cfg_attr(target_arch = "bpf", no_main)]
 
-use crate::error::Error;
-use aya_ebpf::{
-    bindings::xdp_action,
-    helpers::bpf_xdp_adjust_head,
-    macros::{map, xdp},
-    maps::{HashMap, PerCpuArray},
-    programs::XdpContext,
-};
-use aya_log_ebpf::*;
-use channel_data::CdHdr;
-use checksum::ChecksumUpdate;
-use core::net::{Ipv4Addr, Ipv6Addr};
-use ebpf_shared::{
-    ClientAndChannelV4, ClientAndChannelV6, InterfaceAddressV4, InterfaceAddressV6, PortAndPeerV4,
-    PortAndPeerV6,
-};
-use error::SupportedChannel;
-use network_types::{
-    eth::{EthHdr, EtherType},
-    ip::{IpProto, Ipv4Hdr, Ipv6Hdr},
-    udp::UdpHdr,
-};
-use ref_mut_at::ref_mut_at;
+// For non-BPF targets: provide a stub main that exits with an error
+#[cfg(not(target_arch = "bpf"))]
+fn main() {
+    eprintln!("Error: This program is meant to be compiled as an eBPF program.");
+    eprintln!("Use --target bpf or --target bpfel-unknown-none to compile for eBPF.");
+    std::process::exit(1);
+}
 
+// Include modules only for BPF target
+#[cfg(target_arch = "bpf")]
```

Comment:
To make this a bit simpler, we could declare some of these modules inside `try_handle_turn`, then it is implicit that they are only active when we need the `bpf` target.

Reply:
It is possible, you just need to make a directory `try_handle_turn` and move the files there, the `mod` statements to the top of `try_handle_turn.rs` and adjust the imports. Modules can be declared (arbitrarly?) deep.